### PR TITLE
Support Library as well as Store (part 1 at least)

### PR DESCRIPTION
**Pull request by mikeage**
Implements much of #145 

Radio, podcast, and related are not supported, but this should offer a reasonable first implementation of library support in geemusic

### CODE REVIEW

**Comment by mikeage** [comment]
Note that this PR definitely needs testing with a purchased account to make sure I didn't break anything. I can only confirm that songs, albums, and artists work for free accounts ;-)

**Comment by mikeage** [comment]
rebased, and added two commits:
1. Add an environment variable for forcibly disable the store for registered users (good for testing)
2. Defensive code in case we're still indexing when the first request comes in. Edge case, but you never know (plus, it's quite common while testing)

File: geemusic/intents/selection.py [hunk 1]
```diff
@@ -35,7 +35,10 @@ def play_artist(artist_name):
     stream_url = api.get_stream_url(first_song_id)
 
     thumbnail = api.get_thumbnail(artist['artistArtRef'])
-    speech_text = render_template("play_artist_text", artist=artist['name'])
+    if api.is_subscribed:
+        speech_text = render_template("play_artist_text", artist=artist['name'])
+    else:
+        speech_text = render_template("play_artist_text_library", artist=artist['name'])
```
[the remark this answers is not in the document]
Comment (fergyfresh) [reply geemusic/intents/selection.py]: I like the difference in output text.

File: geemusic/utils/music.py [hunk 2]
```diff
@@ -59,31 +62,77 @@ def index_library(self):
...
+        if self.is_subscribed:
+            search = self._search("artist", name)
 
-        if len(search) == 0:
-            return False
+            if len(search) == 0:
+                return False
+
+            return self._api.get_artist_info(search[0]['artistId'],
+                                             max_top_tracks=100)
```
[the remark this answers is not in the document]
Comment (fergyfresh) [reply geemusic/utils/music.py]: That was just for limiting the stream from google servers to support the alexa request from timing out. If it #worksforyou then we can leave that without a limit.

File: geemusic/utils/music.py [hunk 3]
```diff
@@ -59,31 +62,77 @@ def index_library(self):
...
+            if score < 70:
+                return None
+            for song_id, song in self.library.items():
+                if 'artist' in song and song['artist'].lower() == likely_artist.lower():
+                    if not search['topTracks']:  # First entry
+                        # Copy artist details from the first song into the general artist response
+                        search['artistArtRef'] = song['artistArtRef'][0]['url']
+                        search['name'] = song['artist']
+                    search['topTracks'].append(song)
+            random.shuffle(search['topTracks'])  # This is all music, not top, but the user probably would prefer it shuffled.
```
[the remark this answers is not in the document]
Comment (fergyfresh) [reply geemusic/utils/music.py]: It's not the worst implementation in the world and I don't use this feature so its good enough for now.

File: geemusic/utils/music.py [hunk 4]
```diff
@@ -228,8 +297,10 @@ def get_latest_artist_albums(self, artist_name):
     def closest_match(self, request_name, all_matches, artist_name='', minimum_score=70):
         # Give each match a score based on its similarity to the requested
         # name
-        self.log('Fetching library...')
-        
+        self.log('Finding closest match...')
```
[the remark this answers is not in the document]
Comment (fergyfresh) [reply geemusic/utils/music.py]: It's fine i suppose.

File: geemusic/utils/music.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mikeage) [reply geemusic/utils/music.py]: I actually specifically meant the shuffle. The rest seems as logical as I could think of

File: geemusic/utils/music.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikeage) [reply geemusic/utils/music.py]: Gotcha. It seems like it’s not relevant for local, although my biggest test was only about 150 songs.

**Comment by fergyfresh** [comment]
can you add this to it too. somehow you broke the english yaml
https://github.com/fergyfresh/geemusic/commit/e626a7d833892f51c1731841bd63a0cf2f0c0b52

**Comment by fergyfresh** [comment]
https://github.com/fergyfresh/geemusic/tree/18a859563f2ea1f2b8df1c2fa25b5cf7aa6e7771

This is the last setup to work.

**Comment by mikeage** [comment]
That's really strange, since none of this code seems to have changed. See https://github.com/stevenleeg/geemusic/blame/master/geemusic/templates/en.yaml . Perhaps it's related to the PyYAML upgrade? I will look into today

**Comment by mikeage** [comment]
It's just this one, right? Not any other templates?

**Comment by mikeage** [comment]
in any case, I presume this should be on the tip of master, not this branch

**Comment by mikeage** [comment]
Hmmm, with a new virtualenv, and this branch + upstream/master (which I'm pushing now to the branch), I was not able to reproduce it. `ask gee music for help` worked just fine

**Comment by fergyfresh** [comment]
Can you roll pyaml back to the version that worked? Then I can get this merged in.

**Comment by mikeage** [comment]
Re: pyyaml, I'm running with the newer 3.13, and I don't have the problem you saw. Any suggestions for how to reproduce and investigate it? Are you on python 3.6 or 3.7?

**Comment by fergyfresh** [comment]
I'm on 3.6. I am referring to the failed build on travis where I have all the environment variables. It was passing up until the pyyaml updates. I have traced it back that far.

**Comment by fergyfresh** [comment]
This is the last commit that was passing https://github.com/fergyfresh/geemusic/commit/18a859563f2ea1f2b8df1c2fa25b5cf7aa6e7771.

And this commit caused it to fail https://github.com/fergyfresh/geemusic/commit/a191e338f18f161e03fe258aa25ff386ba62a690. 

And its been failing ever since.

**Pull request merged**